packages/backend-tasks/src/tasks/TaskWorker.ts

```diff
@@ -32,13 +33,25 @@ const DEFAULT_WORK_CHECK_FREQUENCY = Duration.fromObject({ seconds: 5 });
  * @private
  */
 export class TaskWorker {
+  private readonly counter: Counter;
+  private readonly duration: Histogram;
+
   constructor(
     private readonly taskId: string,
     private readonly fn: TaskFunction,
     private readonly knex: Knex,
     private readonly logger: Logger,
     private readonly workCheckFrequency: Duration = DEFAULT_WORK_CHECK_FREQUENCY,
-  ) {}
+  ) {
+    const meter = metrics.getMeter('default');
+    this.counter = meter.createCounter('backend_task_runs_total', {
```

Comment:
Would it be an idea to drop `backend_`, from the metric names? Not sure it adds that much and we do not have `backend_catalog_` for example

Reply:
hmm yeah, i just took the package name. Wonder if we should have `scheduled_task` or something as the prefix?

Reply:
Note though that this is one of those weird shared modules. Its metrics will live side by side with the actual plugin metrics so we want to somewhat guard against collisions. And there may be many plugins that all have their own instances of this module all on any given worker node 🤔 `backend_tasks_` (with an extra s) so that it's named just like the package isn't all crazy I think; the table it uses is also prefixed, but then even to `backstage_backend_tasks__*`

Reply:
One more thing on naming. The catalog ones started using dots, is that insane? https://github.com/backstage/backstage/blob/8e206d2431978106a019eb46c913aa00b9fb6af3/plugins/catalog-backend/src/processing/DefaultCatalogProcessingEngine.ts#L392 should we think a bit about what the best and most common naming pattern is out there?

Reply:
I wonder if we should include the `pluginId` as a label then to the metrics and push it through the `TaskWorker`?

Reply:
And yeah, the naming is a little awkward. I wanted to keep the same metrics names for the catalog and the prom metrics in the other PR: #20110

Reply:
yes plugin id is probably not bad, however the task ids are probably generally gonna distinguish well enough on their own typically 🤔

i think it might be good to think about whether we should have these same metrics for local task runs as well; could make sense to make the entire metrics tracking as a wrapper around the callback fn so that it doesn't make any difference HOW the fn was invoked / by whom

Reply:
like `fn: decorateWithMetrics(options.fn, { label1: value1 })` or whatnot